### PR TITLE
PS-7226: Post-merge fixes for the LDAP plugin

### PR DESCRIPTION
**Pull request by dutow**
* Improved logging
* Added TLS testcase
* Modified MTR tests to use environment variables
* Removed some dead code
* authentication_ldap_simple_bind_root_pwd value no longer shown
* Code improvements to conform to the c++/mysql coding conventions

### CODE REVIEW

**Comment by dutow** [comment]
Rebased to 8.0.21

File: plugin/auth_ldap/src/plugin_simple.cc [hunk 1]
```diff
@@ -52,11 +52,43 @@ void update_sysvar(THD *, SYS_VAR *var, void *var_ptr, const void *value) {
...
                           str_or_empty(ca_path), str_or_empty(bind_root_dn),
-                          str_or_empty(bind_root_pwd));
+                          str_or_empty(bind_root_pwd_real));
     connPool->debug_info();
   }
 }
 
+static void pwd_real_set(const char *value) {
+  char *v = nullptr;
+  fprintf(stderr, "XXXXXXX: %s\n", value);
```
Comment (percona-ysorokin) [inline plugin/auth_ldap/src/plugin_simple.cc]: Is this intentional or just a debug print?

File: plugin/auth_ldap/src/plugin_simple.cc [hunk 2]
```diff
@@ -52,11 +52,43 @@ void update_sysvar(THD *, SYS_VAR *var, void *var_ptr, const void *value) {
...
+  if (v && strlen(v)) {
+    bind_root_pwd_real = v;
+    bind_root_pwd = my_strdup(PSI_NOT_INSTRUMENTED, "********", MYF(0));
+  } else {
+    bind_root_pwd = my_strdup(PSI_NOT_INSTRUMENTED, "", MYF(0));
+  }
+}
+
+void update_pwd_sysvar(THD *, SYS_VAR *, void *var_ptr, const void *value) {
+  //*(char *)var_ptr = *(char *)const_cast<void *>(value);
```
Comment (percona-ysorokin) [inline plugin/auth_ldap/src/plugin_simple.cc]: commented out code

File: plugin/auth_ldap/tests/mtr/auth_ldap_simple.result [hunk 3]
```diff
@@ -52,19 +52,19 @@ authentication_ldap_simple_group_search_filter	(|(&(objectClass=posixGroup)(memb
...
 authentication_ldap_simple_max_pool_size	1000
-authentication_ldap_simple_server_host	172.18.0.2
+authentication_ldap_simple_server_host	127.0.0.1
 authentication_ldap_simple_server_port	389
 authentication_ldap_simple_ssl	OFF
 authentication_ldap_simple_tls	OFF
 authentication_ldap_simple_user_search_attr	uid
 SET GLOBAL authentication_ldap_simple_server_host = 'badhost';
-connect(localhost,zoidberg,zoidberg,test,13000,/work/build/mysql-test/var/tmp/mysqld.1.sock);
+connect(localhost,zoidberg,zoidberg,test,13000,/home/dutow/workspace/topics/ldap-fixup/80-local/mysql-test/var/tmp/mysqld.1.sock);
```
Comment (percona-ysorokin) [inline plugin/auth_ldap/tests/mtr/auth_ldap_simple.result]: Non-masked local paths in a MTR `.result` file.

File: plugin/auth_ldap/tests/mtr/auth_ldap_simple.result [hunk 4]
```diff
@@ -52,19 +52,19 @@ authentication_ldap_simple_group_search_filter	(|(&(objectClass=posixGroup)(memb
...
 ERROR 28000: Access denied for user 'zoidberg'@'localhost' (using password: YES)
-SET GLOBAL authentication_ldap_simple_server_host = '172.18.0.2';
+SET GLOBAL authentication_ldap_simple_server_host = '127.0.0.1';;
```
Comment (percona-ysorokin) [inline plugin/auth_ldap/tests/mtr/auth_ldap_simple.result]: Probably `127.0.0.1` should be masked as well as it depends on `$MTR_LDAP_HOST` env variable.
Double semicolon;

File: plugin/auth_ldap/tests/mtr/auth_ldap_simple.result [hunk 5]
```diff
@@ -52,19 +52,19 @@ authentication_ldap_simple_group_search_filter	(|(&(objectClass=posixGroup)(memb
...
 SET GLOBAL authentication_ldap_simple_server_port = 388;
-connect(localhost,zoidberg,zoidberg,test,13000,/work/build/mysql-test/var/tmp/mysqld.1.sock);
+connect(localhost,zoidberg,zoidberg,test,13000,/home/dutow/workspace/topics/ldap-fixup/80-local/mysql-test/var/tmp/mysqld.1.sock);
 ERROR 28000: Access denied for user 'zoidberg'@'localhost' (using password: YES)
-SET GLOBAL authentication_ldap_simple_server_port = 389;
+SET GLOBAL authentication_ldap_simple_server_port = 389;;
```
Comment (percona-ysorokin) [inline plugin/auth_ldap/tests/mtr/auth_ldap_simple.result]: Probably `389` should be masked as well as it depends on `$MTR_LDAP_PORT` env variable.
Double semicolon.

File: plugin/auth_ldap/tests/mtr/auth_ldap_simple.result [hunk 6]
```diff
@@ -52,19 +52,19 @@ authentication_ldap_simple_group_search_filter	(|(&(objectClass=posixGroup)(memb
...
 authentication_ldap_simple_user_search_attr	uid
 SET GLOBAL authentication_ldap_simple_server_host = 'badhost';
-connect(localhost,zoidberg,zoidberg,test,13000,/work/build/mysql-test/var/tmp/mysqld.1.sock);
+connect(localhost,zoidberg,zoidberg,test,13000,/home/dutow/workspace/topics/ldap-fixup/80-local/mysql-test/var/tmp/mysqld.1.sock);
 ERROR 28000: Access denied for user 'zoidberg'@'localhost' (using password: YES)
-SET GLOBAL authentication_ldap_simple_server_host = '172.18.0.2';
+SET GLOBAL authentication_ldap_simple_server_host = '127.0.0.1';;
 SET GLOBAL authentication_ldap_simple_server_port = 388;
-connect(localhost,zoidberg,zoidberg,test,13000,/work/build/mysql-test/var/tmp/mysqld.1.sock);
+connect(localhost,zoidberg,zoidberg,test,13000,/home/dutow/workspace/topics/ldap-fixup/80-local/mysql-test/var/tmp/mysqld.1.sock);
```
Comment (percona-ysorokin) [inline plugin/auth_ldap/tests/mtr/auth_ldap_simple.result]: Non-masked local paths in a MTR .result file.

File: plugin/auth_ldap/tests/mtr/auth_ldap_simple.result [hunk 7]
```diff
@@ -77,7 +77,7 @@ authentication_ldap_simple_group_search_filter	(|(&(objectClass=posixGroup)(memb
 authentication_ldap_simple_init_pool_size	10
 authentication_ldap_simple_log_status	1
 authentication_ldap_simple_max_pool_size	1000
-authentication_ldap_simple_server_host	172.18.0.2
+authentication_ldap_simple_server_host	127.0.0.1
```
Comment (percona-ysorokin) [inline plugin/auth_ldap/tests/mtr/auth_ldap_simple.result]: Should be masked as well

File: plugin/auth_ldap/tests/mtr/auth_ldap_simple.test [hunk 8]
```diff
@@ -6,8 +6,8 @@
 SELECT PLUGIN_NAME, PLUGIN_STATUS FROM INFORMATION_SCHEMA.PLUGINS WHERE PLUGIN_NAME LIKE 'authentication_ldap_simple%';
 SHOW GLOBAL VARIABLES LIKE 'authentication_ldap_simple%';
 SET GLOBAL authentication_ldap_simple_bind_base_dn = 'ou=people,dc=planetexpress,dc=com';
-SET GLOBAL authentication_ldap_simple_server_host = '172.18.0.2';
-SET GLOBAL authentication_ldap_simple_server_port = 389;
+--eval SET GLOBAL authentication_ldap_simple_server_host = '$MTR_LDAP_HOST'
```
Comment (percona-ysorokin) [inline plugin/auth_ldap/tests/mtr/auth_ldap_simple.test:13]: Add `--replace-result $MTR_LDAP_HOST <MTR_LDAP_HOST>`
and 
`--replace-result $MTR_LDAP_PORT <MTR_LDAP_PORT>`
here

File: plugin/auth_ldap/tests/mtr/auth_ldap_simple.test [hunk 9]
```diff
@@ -26,12 +26,12 @@ SET GLOBAL authentication_ldap_simple_server_host = 'badhost';
 --connect (con1,localhost,zoidberg,zoidberg,,,,CLEARTEXT)
 
 
-SET GLOBAL authentication_ldap_simple_server_host = '172.18.0.2';
+--eval SET GLOBAL authentication_ldap_simple_server_host = '$MTR_LDAP_HOST';
```
Comment (percona-ysorokin) [inline plugin/auth_ldap/tests/mtr/auth_ldap_simple.test]: Likewise

File: plugin/auth_ldap/tests/mtr/auth_ldap_simple.test [hunk 10]
```diff
@@ -26,12 +26,12 @@ SET GLOBAL authentication_ldap_simple_server_host = 'badhost';
...
 SET GLOBAL authentication_ldap_simple_server_port = 388;
 --error ER_ACCESS_DENIED_ERROR
 --connect (con1,localhost,zoidberg,zoidberg,,,,CLEARTEXT)
 
-SET GLOBAL authentication_ldap_simple_server_port = 389;
+--eval SET GLOBAL authentication_ldap_simple_server_port = $MTR_LDAP_PORT;
```
Comment (percona-ysorokin) [inline plugin/auth_ldap/tests/mtr/auth_ldap_simple.test]: Likewise

File: plugin/auth_ldap/tests/mtr/auth_ldap_tls.test [hunk 11]
```diff
@@ -0,0 +1,50 @@
+
+# This testcase requires the following ldap structure:
+# https://github.com/rroemhild/docker-test-openldap
+# The easiest way to run it is to use the provided docker image
+# The test also requires the LDAPTLS_REQCERT environment variable set to never
+
+SELECT PLUGIN_NAME, PLUGIN_STATUS FROM INFORMATION_SCHEMA.PLUGINS WHERE PLUGIN_NAME LIKE 'authentication_ldap_simple%';
+SHOW GLOBAL VARIABLES LIKE 'authentication_ldap_simple%';
+SET GLOBAL authentication_ldap_simple_bind_base_dn = 'ou=people,dc=planetexpress,dc=com';
+--eval SET GLOBAL authentication_ldap_simple_server_host = '$MTR_LDAP_HOST'
```
Comment (percona-ysorokin) [inline plugin/auth_ldap/tests/mtr/auth_ldap_tls.test:14]: Add `--replace-result` here

File: plugin/auth_ldap/tests/mtr/auth_ldap_tls.test [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+--eval SET GLOBAL authentication_ldap_simple_server_port = $MTR_LDAP_PORT
```
Comment (percona-ysorokin) [inline plugin/auth_ldap/tests/mtr/auth_ldap_tls.test:16]: Add `--replace-result` here

File: plugin/auth_ldap/tests/mtr/auth_ldap_tls.test [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+
+--disconnect con1
+--connection default
+
+SET GLOBAL authentication_ldap_simple_server_host = 'badhost';
+--error ER_ACCESS_DENIED_ERROR
+--connect (con1,localhost,zoidberg,zoidberg,,,,CLEARTEXT)
+
+
+--eval SET GLOBAL authentication_ldap_simple_server_host = '$MTR_LDAP_HOST';
```
Comment (percona-ysorokin) [inline plugin/auth_ldap/tests/mtr/auth_ldap_tls.test]: Add `--replace-result` here

File: plugin/auth_ldap/tests/mtr/auth_ldap_tls.test [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+SET GLOBAL authentication_ldap_simple_server_port = 388;
+--error ER_ACCESS_DENIED_ERROR
+--connect (con1,localhost,zoidberg,zoidberg,,,,CLEARTEXT)
+
+--eval SET GLOBAL authentication_ldap_simple_server_port = $MTR_LDAP_PORT;
```
Comment (percona-ysorokin) [inline plugin/auth_ldap/tests/mtr/auth_ldap_tls.test]: $MTR_LDAP_PORT

File: plugin/auth_ldap/tests/mtr/auth_ldap_tls.test [hunk 15]
```diff
@@ -0,0 +1,50 @@
+
+# This testcase requires the following ldap structure:
+# https://github.com/rroemhild/docker-test-openldap
+# The easiest way to run it is to use the provided docker image
+# The test also requires the LDAPTLS_REQCERT environment variable set to never
+
```
Comment (percona-ysorokin) [inline plugin/auth_ldap/tests/mtr/auth_ldap_tls.test:6]: Add `count_session.inc` / `wait_until_count_session.inc`

File: plugin/auth_ldap/tests/mtr/auth_ldap_tls.test [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+--disconnect con1
+--connection default
+
+SET GLOBAL authentication_ldap_simple_server_host = 'badhost';
+--error ER_ACCESS_DENIED_ERROR
+--connect (con1,localhost,zoidberg,zoidberg,,,,CLEARTEXT)
+
+
+--eval SET GLOBAL authentication_ldap_simple_server_host = '$MTR_LDAP_HOST';
+SET GLOBAL authentication_ldap_simple_server_port = 388;
```
Comment (percona-ysorokin) [inline plugin/auth_ldap/tests/mtr/auth_ldap_tls.test]: May be
```suggestion
--replace_result ...
SET GLOBAL authentication_ldap_simple_server_port = $MTR_LDAP_HOST + 1;
```

**Review by percona-ysorokin — CHANGES_REQUESTED** [review]

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Comment by dutow** [comment]
@percona-ysorokin I've added an ActiveDirectory test case

**Review by percona-ysorokin — APPROVED** [review]
ActiveDirectory test looks fine

**Pull request merged**